File: api/src/main/java/com/wheats/api/mypage/dto/MyPageProfileResponse.java
package com.wheats.api.mypage.dto;

import com.wheats.api.mypage.entity.UserEntity;

public class MyPageProfileResponse {

    private String name;
    private String email;
    private Integer point;

    public MyPageProfileResponse(String name, String email, Integer point) {
        this.name = name;
        this.email = email;
        this.point = point;
    }

    public static MyPageProfileResponse from(UserEntity user) {
        return new MyPageProfileResponse(
                user.getName(),
                user.getEmail(),
                user.getPoint()
        );
    }

    // ==== Getter만 제공 ====
    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public Integer getPoint() {
        return point;
    }
}

File: mobile/app/src/main/java/com/example/mobile/data/model/MyPageProfileResponse.kt
package com.example.mobile.data.model

/**
 * 내 프로필 응답 DTO
 * - 서버의 MyPageProfileResponse(Java)와 필드 구조를 맞춘 버전
 */
data class MyPageProfileResponse(
    val name: String,
    val email: String,
    val point: Int
)


